crates/runtime/src/context.rs

```diff
@@ -0,0 +1,216 @@
+            16_f64 * STEP_RESOURCE_COST,
+        ),
+        (
+            SIGNATURE_BUILTIN_NAME.to_string(),
+            2048_f64 * STEP_RESOURCE_COST,
+        ), // ECDSA
+        (
+            BITWISE_BUILTIN_NAME.to_string(),
+            64_f64 * STEP_RESOURCE_COST,
+        ),
+        (
+            POSEIDON_BUILTIN_NAME.to_string(),
+            32_f64 * STEP_RESOURCE_COST,
+        ),
+        (OUTPUT_BUILTIN_NAME.to_string(), 0_f64 * STEP_RESOURCE_COST),
+        (
+            EC_OP_BUILTIN_NAME.to_string(),
+            1024_f64 * STEP_RESOURCE_COST,
+        ),
+        (
+            KECCAK_BUILTIN_NAME.to_string(),
+            2048_f64 * STEP_RESOURCE_COST, // 2**11
+        ),
+        (
+            SEGMENT_ARENA_BUILTIN_NAME.to_string(),
+            0_f64 * STEP_RESOURCE_COST,
+        ),
+    ]));
+
+    BlockContext {
+        chain_id: ChainId("SN_GOERLI".to_string()),
+        block_number: block_info.block_number,
+        block_timestamp: block_info.timestamp,
+        sequencer_address: block_info.sequencer_address,
+        vm_resource_fee_cost,
+        invoke_tx_max_n_steps: 3_000_000,
+        validate_max_n_steps: 1_000_000,
+        max_recursion_depth: 50,
+        fee_token_addresses: FeeTokenAddresses {
+            strk_fee_token_address: ContractAddress(patricia_key!(ERC20_CONTRACT_ADDRESS)),
+            eth_fee_token_address: ContractAddress(patricia_key!(ERC20_CONTRACT_ADDRESS)),
+        },
+        gas_prices: GasPrices {
+            eth_l1_gas_price: 100 * u128::pow(10, 9),
+            strk_l1_gas_price: 100 * u128::pow(10, 9),
+        },
+    }
+}
+
+#[must_use]
+pub fn build_transaction_context() -> AccountTransactionContext {
+    AccountTransactionContext::Current(CurrentAccountTransactionContext {
+        common_fields: CommonAccountFields {
+            transaction_hash: TransactionHash::default(),
+            version: TransactionVersion::THREE,
+            signature: TransactionSignature::default(),
+            nonce: Nonce(StarkFelt::from(0_u8)),
+            sender_address: ContractAddress::default(),
+            only_query: false,
+        },
+        resource_bounds: ResourceBoundsMapping(BTreeMap::from([
+            (
+                Resource::L1Gas,
+                ResourceBounds {
+                    max_amount: 0,
+                    max_price_per_unit: 1,
+                },
+            ),
+            (
+                Resource::L2Gas,
+                ResourceBounds {
+                    max_amount: 0,
+                    max_price_per_unit: 0,
+                },
+            ),
+        ])),
+        tip: Default::default(),
+        nonce_data_availability_mode: DataAvailabilityMode::L1,
+        fee_data_availability_mode: DataAvailabilityMode::L1,
+        paymaster_data: Default::default(),
+        account_deployment_data: Default::default(),
+    })
+}
+
+#[derive(Copy, Clone, Serialize, Deserialize, Debug)]
+pub struct BlockInfo {
```

Comment:
Stuff from there to the end of the file should be in a separate file I think, maybe `state.rs`?

Reply:
improved